<launch>
  <arg name="json" default="$(find jsk_2015_05_baxter_apc)/json/layout_12.json"/>
  <arg name="UPPER_SHELF"/>
  <arg name="LOWER_SHELF"/>
  <arg name="TARGET_BIN"/>

  <arg name="INPUT_POINTS"/>
  <arg name="INPUT_INFO"/>
  <arg name="INPUT_IMAGE"/>

  <!-- set ros param -->  
  <node pkg="jsk_apc2016_common" type="publish_bin_info.py"
    name="set_bin_param" output="screen">
    <param name="json" value="$(arg json)"/>
    <rosparam file="$(arg UPPER_SHELF)" command="load" ns="upper_shelf"/>
    <rosparam file="$(arg LOWER_SHELF)" command="load" ns="lower_shelf"/>
  </node>


  <arg name="INPUT_BIN_ARRAY" value="/set_bin_param/bin_array"/>

  <param name="/right_hand/target_bin" value="$(arg TARGET_BIN)"/>
  <arg name="INPUT_TARGET" value="/right_hand/target_bin"/>


  <!-- note: initial_rot_list has been changed from 2015 to be compatiable 
              with Quaterniona-->




  <node name="segmentation_in_bin"
        pkg="jsk_2016_01_baxter_apc" 
        type="rbo_segmentation_in_bin_node.py" 
        output="screen">
    <remap from="~input" to="$(arg INPUT_POINTS)"/>
    <remap from="~input/info" to="$(arg INPUT_INFO)"/>
    <remap from="~input/image" to="$(arg INPUT_IMAGE)"/>
    <remap from="~input/bin_info_array" to="$(arg INPUT_BIN_ARRAY)"/>
    <remap from="~target_bin_name" to="$(arg INPUT_TARGET)"/>
    <param name="trained_pkl_path" value="$(find jsk_apc2016_common)/data/trained_segmenter.pkl" />
  </node>
</launch>
